Build error on post-5.17 kernels

mkosi-mainline started failing with:

```
/root/src/src/modules/exploit_detection/p_exploit_detection.c:503:46: error: unknown type name 'mm_segment_t'
  503 | notrace static inline void p_dump_addr_limit(mm_segment_t *p_addr_limit, struct task_struct *p_task) {
      |                                              ^~~~~~~~~~~~
```

Linux 5.17 had been out for a few more days, and mkosi-mainline builds were not failing yet, so I assume this is actually a post-5.17 change, even though the kernel version number hasn't been updated further yet.

When fixing this, we should be careful not to claim we're fixing a 5.17+ issue, as that would be misleading. I mention this because we, unfortunately, already have the same kind of error in some past commit messages. We can instead write "post-5.17".

--- a/src/modules/exploit_detection/p_exploit_detection.c
+++ b/src/modules/exploit_detection/p_exploit_detection.c
@@ -500,18 +500,16 @@ notrace void p_verify_addr_limit(struct p_ed_process *p_orig, struct task_struct
 #endif
 }
 
-notrace static inline void p_dump_addr_limit(mm_segment_t *p_addr_limit, struct task_struct *p_task) {
-
 #if defined(P_VERIFY_ADDR_LIMIT)
-
+notrace static inline void p_dump_addr_limit(mm_segment_t *p_addr_limit, struct task_struct *p_task) {
 #if defined(CONFIG_X86)
    p_addr_limit->seg =
 #elif defined(CONFIG_ARM) || defined(CONFIG_ARM64)
    *p_addr_limit =
 #endif
                        p_get_addr_limit(p_task);
-#endif
 }
+#endif
 
 notrace void p_update_ed_process(struct p_ed_process *p_source, struct task_struct *p_task, char p_stack) {
 

